scheduler/system_sched.go

```diff
@@ -390,3 +394,28 @@ func (s *SystemScheduler) computePlacements(place []allocTuple) error {
 
 	return nil
 }
+
+// addBlocked creates a new blocked eval for this job on this node
+// - Keep blocked evals in the scheduler, but just for grins
+// - Submits to the planner (worker.go), which keeps the eval for execution later
+func (s *SystemScheduler) addBlocked(node *structs.Node) error {
+	e := s.ctx.Eligibility()
+	escaped := e.HasEscaped()
+
+	// Only store the eligible classes if the eval hasn't escaped.
+	var classEligibility map[string]bool
+	if !escaped {
+		classEligibility = e.GetClasses()
+	}
+
+	blocked := s.eval.CreateBlockedEval(classEligibility, escaped, e.QuotaLimitReached())
+	blocked.StatusDescription = blockedEvalFailedPlacements
+	blocked.NodeID = node.ID
+
+	if s.blocked == nil {
+		s.blocked = map[string]*structs.Evaluation{}
+	}
+	s.blocked[node.ID] = blocked
```

Comment:
I had a harder time following time reading this code due to reusing `blocked` name to refer to the blocked evaluation as well as container of blocked evals.